client/store/reports/utils.js

```diff
@@ -46,6 +46,41 @@ export function getFilterQuery( endpoint, query ) {
 	return {};
 }
 
+/**
+ * Add timestamp to advanced filter parameters involving date. The api
+ * expects a timestamp for these values similar to `before` and `after`.
+ *
+ * @param {object} config - advancedFilters config object.
+ * @param {object} activeFilter - an active filter.
+ * @returns {object} - an active filter with timestamp added to date values.
+ */
+export function timeStampFilterDates( config, activeFilter ) {
+	const advancedFilterConfig = config.filters[ activeFilter.key ];
+	if ( 'Date' === get( advancedFilterConfig, [ 'input', 'component' ] ) ) {
```

Comment:
Instead of having all this logic in the if clause, what do you think about testing for `Date` first, and if falsey, `return activeFilter` - essentially [return early](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md#indentation-and-line-breaks) when possible.

Reply:
Thanks @timmyc. Are you suggesting this?

```js
if ( 'Date' !== get( advancedFilterConfig, [ 'input', 'component' ] ) ) {
    return activeFilter;
}
if ( Array.isArray( value ) ) {
    // do some stuff
   return something;
}

// do some other stuff
return anotherThing;
```